### PR TITLE
[wip]Add telemetry checks in serving downstream integration tests

### PR DESCRIPTION
**Pull request by skonto**
This is a follow up of #657.

### CODE REVIEW

File: test/servinge2e/helpers.go [hunk 1]
```diff
@@ -44,3 +53,128 @@ func withServiceReadyOrFail(ctx *test.Context, service *servingv1.Service) *serv
...
+	routes         float64
+	configurations float64
+	revisions      float64
+}
+
+func getMetricsEndpointPath() string {
+	portAndPath := strconv.Itoa(8383) + "/metrics"
+	metricsPath := "http://knative-openshift-metrics.openshift-serverless:" + portAndPath
+	return metricsPath
+}
```
Comment (markusthoemmes) [inline test/servinge2e/helpers.go:68]: This won't work as the hostname is hardcoded. You'll have to fetch it from the route after it's applied and accepted. The port will also be 80.

File: test/servinge2e/helpers.go [hunk 2]
```diff
@@ -44,3 +53,128 @@ func withServiceReadyOrFail(ctx *test.Context, service *servingv1.Service) *serv
...
+			Path: "/metrics",
+			To: v1.RouteTargetReference{
+				Kind: "Service",
+				Name: "knative-openshift-metrics",
+			},
+			TLS: &v1.TLSConfig{
+				Termination:                   v1.TLSTerminationEdge,
+				InsecureEdgeTerminationPolicy: v1.InsecureEdgeTerminationPolicyAllow,
+			},
+			WildcardPolicy: v1.WildcardPolicyNone,
```
Comment (markusthoemmes) [inline test/servinge2e/helpers.go:167]: Are these required? I'd think no.

File: test/servinge2e/service_to_service_test.go [hunk 3]
```diff
@@ -89,6 +98,53 @@ func testServiceToService(t *testing.T, ctx *test.Context, namespace string, tc
 		t.Errorf("the Route at domain %s didn't serve the expected text %q: %v", service.Status.URL.URL(), helloworldText, err)
 	}
 
+	// Check if service monitor service is available, at this point it should be present
+	_, err := ctx.Clients.Kube.CoreV1().Services("openshift-serverless").Get(context.Background(), "knative-openshift-metrics", meta.GetOptions{})
+	if err != nil {
+		t.Errorf("error getting service monitor service: %v", err)
```
Comment (markusthoemmes) [inline test/servinge2e/service_to_service_test.go:104]: This should be Fatal as the rest of the test won't work (actually true for almost all statements hereafter)

File: test/servinge2e/servicemesh_test.go [hunk 4]
```diff
@@ -37,10 +37,11 @@ import (
...
 type testCase struct {
-	name               string
-	labels             map[string]string // Ksvc labels
-	annotations        map[string]string // Revision template annotations
-	expectIstioSidecar bool              // Whether it is expected for the istio-proxy sidecar to be injected into the pod
+	name                string
+	labels              map[string]string // Ksvc labels
+	annotations         map[string]string // Revision template annotations
+	expectIstioSidecar  bool              // Whether it is expected for the istio-proxy sidecar to be injected into the pod
+	expectedMetricCount float64
```
Comment (markusthoemmes) [inline test/servinge2e/servicemesh_test.go:44]: Let's please create an extra test for this. This has nothing to do with the Service to Service tests and will just make picking apart test results harder as the service-to-service tests might fail due to this.

File: test/servinge2e/helpers.go [hunk 5]
```diff
@@ -44,3 +53,128 @@ func withServiceReadyOrFail(ctx *test.Context, service *servingv1.Service) *serv
...
+}
+
+func getMetricsEndpointPath() string {
+	portAndPath := strconv.Itoa(8383) + "/metrics"
+	metricsPath := "http://knative-openshift-metrics.openshift-serverless:" + portAndPath
+	return metricsPath
+}
+func fetchTelemetryMetrics() (*telemetryStat, error) {
+	metricsPath := getMetricsEndpointPath()
+	req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, metricsPath, nil)
```
Comment (markusthoemmes) [inline test/servinge2e/helpers.go:71]: nit: NewRequest is just as fine if you use a background context anyway.

File: test/servinge2e/helpers.go [hunk 6]
```diff
@@ -44,3 +53,128 @@ func withServiceReadyOrFail(ctx *test.Context, service *servingv1.Service) *serv
...
+	if err != nil {
+		return nil, err
+	}
+	httpClient := http.Client{}
+	resp, err := httpClient.Do(req)
```
Comment (markusthoemmes) [inline test/servinge2e/helpers.go:76]: The response's body needs to be closed.

File: test/servinge2e/servicemesh_test.go (hunk 4 above)
Comment (skonto) [reply test/servinge2e/servicemesh_test.go:44]: In the past we discussed that I should not create a separate test to save some time. I guess this kind of test which just sets the admin does not cost anything right?

File: test/servinge2e/helpers.go (hunk 1 above)
Comment (skonto) [reply test/servinge2e/helpers.go:68]: Yeah I need the domain.

File: test/servinge2e/servicemesh_test.go (hunk 4 above)
Comment (markusthoemmes) [reply test/servinge2e/servicemesh_test.go:44]: Well we just need to create a Knative Service and then check the stuff you're checking. That might cost a second or two more, but that's absolutely worth it here.

We wouldn't want to install Serving separately for it though, yeah. We should be good here with a separate test in the same suite.

File: test/servinge2e/helpers.go (hunk 2 above)
Comment (skonto) [reply test/servinge2e/helpers.go:167]: I guess they are meant to allow non-secure connections within the cluster and so taht tls is terminated to the edge, I guess not relevant if we are hitting with http only.

File: test/servinge2e/service_to_service_test.go (hunk 3 above)
Comment (skonto) [reply test/servinge2e/service_to_service_test.go:104]: Yeah I was thinking about it guess if I move this in a separate test it makes sense.

**Comment by skonto** [comment]
@markusthoemmes @bbrowning as we discussed this is going to be provided by OCP in future release 4.7+. I will close this one and also revert #657

**Pull request closed without merging**